pkg/http/errors.go

```diff
@@ -0,0 +1,47 @@
+package http
+
+import (
+	"fmt"
+	"time"
+)
+
+type HttpError struct {
+	httpStatus int
+
+	Code    string `json:"code"`
+	Message string `json:"message"`
+	Line    int    `json:"line"`
```

Comment:
The "line" field is optional:
```suggestion
	Line    int    `json:"line,omitempty"`
```